commit e9f37e27a0db5f2026e5ab33e47faf4e558050af
Author: Fabrice Bellet <fabrice@bellet.info>
Date:   Thu Oct 23 23:48:57 2008 -0700

    iwl3945: iwl3945_mac_get_tsf() should not return zero
    
    The problem fixed here is that iwl3945_mac_get_tsf() returns 0, as the
    function is not implemented, and this is considered as a valid value by
    the mac layer in mlme.c:1605. The consequence is that the STA in ad-hoc
    mode is inserted/removed quite frequently due to IBSS merging.
    
    This patch fixes :
    http://www.intellinuxwireless.org/bugzilla/show_bug.cgi?id=1781
    and https://bugzilla.redhat.com/show_bug.cgi?id=459401
    
    Signed-off-by: Fabrice Bellet <fabrice@bellet.info>
    Acked-by: Tomas Winkler <tomas.winkler@intel.com>
    Signed-off-by: Reinette Chatre <reinette.chatre@intel.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

(Rebased to match current Rawhide kernel... -- JWL)

diff -up linux-2.6.27.noarch/drivers/net/wireless/iwlwifi/iwl3945-base.c.orig linux-2.6.27.noarch/drivers/net/wireless/iwlwifi/iwl3945-base.c
--- linux-2.6.27.noarch/drivers/net/wireless/iwlwifi/iwl3945-base.c.orig	2008-10-09 18:13:53.000000000 -0400
+++ linux-2.6.27.noarch/drivers/net/wireless/iwlwifi/iwl3945-base.c	2008-10-27 17:41:59.000000000 -0400
@@ -7164,14 +7164,6 @@ static int iwl3945_mac_get_stats(struct 
 	return 0;
 }
 
-static u64 iwl3945_mac_get_tsf(struct ieee80211_hw *hw)
-{
-	IWL_DEBUG_MAC80211("enter\n");
-	IWL_DEBUG_MAC80211("leave\n");
-
-	return 0;
-}
-
 static void iwl3945_mac_reset_tsf(struct ieee80211_hw *hw)
 {
 	struct iwl3945_priv *priv = hw->priv;
@@ -7828,7 +7820,6 @@ static struct ieee80211_ops iwl3945_hw_o
 	.get_stats = iwl3945_mac_get_stats,
 	.get_tx_stats = iwl3945_mac_get_tx_stats,
 	.conf_tx = iwl3945_mac_conf_tx,
-	.get_tsf = iwl3945_mac_get_tsf,
 	.reset_tsf = iwl3945_mac_reset_tsf,
 	.hw_scan = iwl3945_mac_hw_scan
 };
